### ISSUE TITLE
Output log for python/matlab

### ISSUE THREAD

**Issue by rijobro** [CCPPETMR/SIRF#332]
I know this is extremely late in the day to be throwing in suggestions for SIRF v2. However, I think debugging in Matlab and Python would be greatly improved if we had a logging process. My C++ errors that propogate through are out of sync and often truncated, such that I can't see the useful information.

Thoughts?

**Comment by KrisThielemans** [comment]
sure, but this wouldn't break backwards compatibility, so we can call this 2.1.

**Comment by KrisThielemans** [comment]
it isn't trivial to capture this. @evgueni-ovtchinnikov had to work hard to get STIR's `info` etc captured. (which is the `MessageRedirector` stuff)

**Comment by rijobro** [comment]
Yes, it certainly seems like a tricky task (but one that would be extremely useful I feel). Ok to wait for v2.1.
